### PR TITLE
[Docs] Refine Cloud Tutorials

### PR DESCRIPTION
**Pull request by yinweisu**
*Issue #, if available:*

*Description of changes:*
* Added info on how to train/deploy/batch inference with other predictor types.
* Updated the doc to reflect latest changes.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/tutorials/cloud_fit_deploy/cloud-aws-sagemaker-deployment.md [hunk 1]
```diff
@@ -101,17 +114,16 @@ model_data = sagemaker_session.upload_data(
...
 ```{.python}
-instance_type = "ml.m5.2xlarge"
+instance_type = "ml.m5.2xlarge"  # You might want to use GPU instances for Text/Image/MultiModal Predictors etc
 
 model = AutoGluonInferenceModel(
     model_data=model_data,
     role=role,
     region=region,
-    framework_version="0.4",
+    framework_version="0.5.2",  # Replace this with the AutoGLuon DLC container version you want to use
```
Comment (Innixma) [inline docs/tutorials/cloud_fit_deploy/cloud-aws-sagemaker-deployment.md:123]: typo: AutoGluon

File: docs/tutorials/cloud_fit_deploy/cloud-aws-sagemaker-deployment.md [hunk 2]
```diff
@@ -101,17 +114,16 @@ model_data = sagemaker_session.upload_data(
...
     py_version="py38",
     instance_type=instance_type,
-    source_dir="scripts",
-    entry_point="tabular_serve.py",
+    entry_point="YOUR_SERVING_SCRIPT_PATH",
```
Comment (Innixma) [inline docs/tutorials/cloud_fit_deploy/cloud-aws-sagemaker-deployment.md:126]: Add: `  # example: "tabular_serve.py" `

File: docs/tutorials/cloud_fit_deploy/cloud-aws-sagemaker-deployment.md [hunk 3]
```diff
@@ -148,17 +162,16 @@ model_data = sagemaker_session.upload_data(
...
     model_data=model_data,
     role=role,
     region=region,
-    framework_version="0.4",
+    framework_version="0.5.2",  # Replace this with the AutoGLuon DLC container version you want to use
     py_version="py38",
     instance_type=instance_type,
-    entry_point="tabular_serve-batch.py",
-    source_dir="scripts",
+    entry_point="YOUR_BATCH_SERVE_SCRIPT",
```
Comment (Innixma) [inline docs/tutorials/cloud_fit_deploy/cloud-aws-sagemaker-deployment.md:174]: ditto example comment

File: docs/tutorials/cloud_fit_deploy/cloud-aws-sagemaker-training.md [hunk 4]
```diff
@@ -118,8 +118,20 @@ if __name__ == "__main__":
             lb = predictor.leaderboard(silent=False)
             lb.to_csv(f"{args.output_data_dir}/leaderboard.csv")
 ```
+For training other types of AutoGluon Predictors, i.e. TextPredictor, the training script you provided will be quite similar to the one above.
+Mostly, you just need to replace `TabularPredictor` to be `TextPredictor` for example.
+Keep in mind that the specific Predictor type you want to train might not support the same feature sets as `TabularPredictor`.
+For example, `leaderboard` does not exists for all Predictors.
```
Comment (Innixma) [inline docs/tutorials/cloud_fit_deploy/cloud-aws-sagemaker-training.md:124]: typo: exist

File: docs/tutorials/cloud_fit_deploy/cloud-aws-sagemaker-training.md [hunk 5]
```diff
@@ -118,8 +118,20 @@ if __name__ == "__main__":
...
 
-YAML config:
+
+### Notes for Training
+1. If your use case involves image modality, you will need to pass the images as a compressed file to the training container(similarly to how you would pass in train data), decompress the file in the training container, and update the training data columns with the updated image path in the container.
```
Comment (Innixma) [inline docs/tutorials/cloud_fit_deploy/cloud-aws-sagemaker-training.md:128]: `training container(similarly ` missing space

**Review by Innixma — APPROVED** [review]
LGTM! Added some minor comments

File: docs/tutorials/cloud_fit_deploy/cloud-aws-sagemaker-deployment.md [hunk 6]
```diff
@@ -59,10 +64,16 @@ def transform_fn(model, request_body, input_content_type, output_content_type="a
...
     pred = model.predict(data)
     pred_proba = model.predict_proba(data)
-    prediction = pd.concat([pred, pred_proba], axis=1).values
+    prediction = pd.concat([pred, pred_proba], axis=1)
 
-    return json.dumps(prediction.tolist()), output_content_type
+    return prediction.to_json(), output_content_type
 ```
+For inference with other types of AutoGluon Predictors, i.e. TextPredictor, the inference script you provided will be quite similar to the one above.
+Mostly, you just need to replace `TabularPredictor` to be `TextPredictor` for example.
```
Comment (zhiqiangdon) [inline docs/tutorials/cloud_fit_deploy/cloud-aws-sagemaker-deployment.md:72]:
```
globals()["column_names"] = model.feature_metadata_in.get_features()
model.persist_models()
```
I guess this code doesn't work for `MultiModalPredictor`. Could we add if check in `model_fn`?

File: docs/tutorials/cloud_fit_deploy/cloud-aws-sagemaker-deployment.md [hunk 7]
```diff
@@ -59,10 +64,16 @@ def transform_fn(model, request_body, input_content_type, output_content_type="a
...
+
+### Note on image modality
+To do inference on image modality, you would need to embed the image info, as bytes for example, into a column of the test data.
+Then in the inference container, you would need to store the image into the disk and update the test data with the image paths accordingly.
```
Comment (zhiqiangdon) [inline docs/tutorials/cloud_fit_deploy/cloud-aws-sagemaker-deployment.md:76]: This description is kind of too high-level. Could we add some code examples?

File: docs/tutorials/cloud_fit_deploy/cloud-aws-sagemaker-deployment.md [hunk 8]
```diff
@@ -36,6 +36,9 @@ import numpy as np
 def model_fn(model_dir):
     """loads model from previously saved artifact"""
     model = TabularPredictor.load(model_dir)
```
Comment (zhiqiangdon) [inline docs/tutorials/cloud_fit_deploy/cloud-aws-sagemaker-deployment.md:38]: Could we add `# or model = MultiModalPredictor.load(model_dir)`?

File: docs/tutorials/cloud_fit_deploy/cloud-aws-sagemaker-deployment.md [hunk 9]
```diff
@@ -148,17 +162,16 @@ model_data = sagemaker_session.upload_data(
...
     role=role,
     region=region,
-    framework_version="0.4",
+    framework_version="0.5.2",  # Replace this with the AutoGLuon DLC container version you want to use
     py_version="py38",
     instance_type=instance_type,
-    entry_point="tabular_serve-batch.py",
-    source_dir="scripts",
+    entry_point="YOUR_BATCH_SERVE_SCRIPT",
     predictor_cls=AutoGluonTabularPredictor,
```
Comment (zhiqiangdon) [inline docs/tutorials/cloud_fit_deploy/cloud-aws-sagemaker-deployment.md:175]: Could we add `# or AutoGluonMultiModalPredictor if model is trained by MultiModalPredictor`?

File: docs/tutorials/cloud_fit_deploy/cloud-aws-sagemaker-deployment.md [hunk 10]
```diff
@@ -148,17 +162,16 @@ model_data = sagemaker_session.upload_data(
 Prepare transform job:
 
 ```{.python}
-instance_type = "ml.m5.2xlarge"
+instance_type = "ml.m5.2xlarge"  # You might want to use GPU instances for Text/Image/MultiModal Predictors etc
```
Comment (zhiqiangdon) [inline docs/tutorials/cloud_fit_deploy/cloud-aws-sagemaker-deployment.md:165]: Could we give an GPU instance example? Maybe a G4 instance.

File: docs/tutorials/cloud_fit_deploy/cloud-aws-sagemaker-deployment.md [hunk 11]
```diff
@@ -200,15 +209,23 @@ import numpy as np
 def model_fn(model_dir):
     """loads model from previously saved artifact"""
     model = TabularPredictor.load(model_dir)
+    # Get original columns for header validation later
+    # Different predictors might have different ways to retrieve this info
+    # Free to remove this check if you know your test data is always valid.
     globals()["column_names"] = model.feature_metadata_in.get_features()
+    model.persist_models()
```
Comment (zhiqiangdon) [inline docs/tutorials/cloud_fit_deploy/cloud-aws-sagemaker-deployment.md:216]: Ditto. This code may not work with `MultiModalPredictor`.

File: docs/tutorials/cloud_fit_deploy/cloud-aws-sagemaker-deployment.md [hunk 12]
```diff
@@ -16,7 +16,7 @@ To start using the containers, an inference script and the [wrapper classes](htt
 When authoring an inference [scripts](https://github.com/aws/amazon-sagemaker-examples/blob/master/advanced_functionality/autogluon-tabular-containers/scripts/), 
 please refer to SageMaker [documentation](https://docs.aws.amazon.com/sagemaker/latest/dg/adapt-inference-container.html).
 
-Here is one of the possible inference scripts. 
+Here is one of the possible Tabular inference scripts.
```
Comment (zhiqiangdon) [inline docs/tutorials/cloud_fit_deploy/cloud-aws-sagemaker-deployment.md:19]: Do we need to mention `Tabular` here? Is this tutorial mainly for tabular predictor deployment or general for all predictors? I guess the later one.

File: docs/tutorials/cloud_fit_deploy/cloud-aws-sagemaker-deployment.md [hunk 13]
```diff
@@ -36,6 +36,9 @@ import numpy as np
 def model_fn(model_dir):
```
Comment (zhiqiangdon) [inline docs/tutorials/cloud_fit_deploy/cloud-aws-sagemaker-deployment.md:36]: Could we add `# from autogluon.multimodal import MultiModalPredictor` in the imports?

File: docs/tutorials/cloud_fit_deploy/cloud-aws-sagemaker-deployment.md (hunk 6 above)
Comment (yinweisu) [reply docs/tutorials/cloud_fit_deploy/cloud-aws-sagemaker-deployment.md:72]: Yes, but I did add a comment there to remind the user:
```
# Get original columns for header validation later
# Different predictors might have different ways to retrieve this info
# Free to remove this check if you know your test data is always valid.
```

File: docs/tutorials/cloud_fit_deploy/cloud-aws-sagemaker-deployment.md (hunk 12 above)
Comment (yinweisu) [reply docs/tutorials/cloud_fit_deploy/cloud-aws-sagemaker-deployment.md:19]: The tutorial is not only for `Tabular` but this example script is. Basically the idea here is that users can use this example script as a starting point and make some modification to apply on other predictors. I've mentioned it in this way:
```
For inference with other types of AutoGluon Predictors, i.e. TextPredictor, the inference script you provided will be quite similar to the one above.
Mostly, you just need to replace `TabularPredictor` to be `TextPredictor` for example.
```

File: docs/tutorials/cloud_fit_deploy/cloud-aws-sagemaker-deployment.md (hunk 7 above)
Comment (yinweisu) [reply docs/tutorials/cloud_fit_deploy/cloud-aws-sagemaker-deployment.md:76]: Good suggestion!

File: docs/tutorials/cloud_fit_deploy/cloud-aws-sagemaker-deployment.md (hunk 10 above)
Comment (yinweisu) [reply docs/tutorials/cloud_fit_deploy/cloud-aws-sagemaker-deployment.md:165]: good call

**Review by zhiqiangdon — APPROVED** [review]
LGTM! Thanks for the multimodal support! Adding more comments in the code will help improve the usability.

**Pull request merged**